pkg/cmd/roachtest/roachtestutil/mixedversion/planner.go

```diff
@@ -1362,6 +1363,23 @@ func (ss stepSelector) Filter(predicate func(*singleStep) bool) stepSelector {
 	return result
 }
 
+func (ss stepSelector) Cut(predicate func(*singleStep) bool) (stepSelector, stepSelector) {
```

Comment:
Lets leave a comment explaining this function. What does it do/return? Is the step where the predicate was true part of the before, after, or removed completely?